include/hip/hcc_detail/device_functions.h

```diff
@@ -748,26 +748,98 @@ unsigned long long int __ballot64(int predicate) {
 // hip.amdgcn.bc - lanemask
 __device__
 inline
-int64_t  __lanemask_gt()
+uint64_t  __lanemask_gt()
 {
-    int32_t activelane = __ockl_activelane_u32();
-    int64_t ballot = __ballot64(1);
-    if (activelane != 63) {
-        int64_t tmp = (~0ULL) << (activelane + 1);
-        return tmp & ballot;
-    }
-    return 0;
+    uint32_t lane = __ockl_lane_u32();
+    if (lane == 63)
+      return 0;
+    uint64_t ballot = __ballot64(1);
+    uint64_t mask = (~((uint64_t)0)) << (lane + 1);
+    return mask & ballot;
 }
 
 __device__
 inline
-int64_t __lanemask_lt()
+uint64_t __lanemask_lt()
 {
-    int32_t activelane = __ockl_activelane_u32();
+    uint32_t lane = __ockl_lane_u32();
     int64_t ballot = __ballot64(1);
-    if (activelane == 0)
-        return 0;
-    return ballot;
+    uint64_t mask = ((uint64_t)1 << lane) - (uint64_t)1;
+    return mask & ballot;
+}
+
+__device__
+inline
+uint64_t  __lanemask_eq()
+{
+    uint32_t lane = __ockl_lane_u32();
+    int64_t mask = ((uint64_t)1 << lane);
+    return mask;
+}
+
+__device__
+inline
+uint64_t __activemask()
+{
+    return __builtin_amdgcn_read_exec();
+}
+
+__device__
+inline
+uint32_t __lane()
+{
+    return __ockl_lane_u32();
+}
+
+/**
+ * grid and multi-grid cooperative groups related device library API wrappers
```

Comment:
I'm not sure these need to be here.  Suggest calling OCKL functions directly from helper for the time being.